src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/scheduling/GlobalConfChecker.java

```diff
@@ -0,0 +1,265 @@
+import ee.ria.xroad.common.conf.serverconf.model.ServerConfType;
+import ee.ria.xroad.common.identifier.ClientId;
+import ee.ria.xroad.common.identifier.SecurityServerId;
+import ee.ria.xroad.common.util.CertUtils;
+import ee.ria.xroad.commonui.SignerProxy;
+import ee.ria.xroad.signer.protocol.SignerClient;
+import ee.ria.xroad.signer.protocol.dto.AuthKeyInfo;
+import ee.ria.xroad.signer.protocol.dto.CertificateInfo;
+import ee.ria.xroad.signer.protocol.dto.KeyUsageInfo;
+import ee.ria.xroad.signer.protocol.message.GetAuthKey;
+
+import lombok.extern.slf4j.Slf4j;
+import org.niis.xroad.restapi.facade.GlobalConfFacade;
+import org.quartz.JobExecutionException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.security.cert.X509Certificate;
+
+import static ee.ria.xroad.common.ErrorCodes.translateException;
+import static ee.ria.xroad.common.SystemProperties.NodeType.SLAVE;
+import static ee.ria.xroad.common.util.CryptoUtils.readCertificate;
+
+/**
+ * Job that checks whether globalconf has changed.
+ */
+@Component
+@Transactional
+@Slf4j
+public class GlobalConfChecker {
+    public static final int JOB_REPEAT_INTERVAL_MS = 30000;
+    public static final int INITIAL_DELAY_MS = 30000;
+    @Autowired
+    private GlobalConfCheckerHelper globalConfCheckerHelper;
+    @Autowired
+    private GlobalConfFacade globalConfFacade;
+
+    /**
+     * Reloads global configuration, and updates client statuses, authentication certificate statuses
+     * and server owner identity to the serverconf database. The task is scheduled at a fixed rate
+     * which means that the task is run at a fixed interval (defined by FIXED_RATE_MS) regardless of the
+     * previous executions of the task. However, scheduled tasks do not run in parallel by default. The
+     * next task won't be invoked until the previous one is done. Set an initial delay before running the task
+     * for the first time after a startup to be sure that all required components are available, e.g.
+     * SignerClient may not be available immediately.
+     * @throws JobExecutionException
+     */
+    @Scheduled(fixedRate = JOB_REPEAT_INTERVAL_MS, initialDelay = INITIAL_DELAY_MS)
+    public void updateServerConf() throws JobExecutionException {
+        // In clustered setup slave nodes may skip globalconf updates
+        if (SLAVE.equals(SystemProperties.getServerNodeType())) {
+            log.debug("This is a slave node - skip globalconf updates");
+            return;
+        }
+
+        try {
+            log.debug("Check globalconf for updates");
+            checkGlobalConf();
+        } catch (Exception e) {
+            log.error("Checking globalconf for updates failed", e);
+            throw new JobExecutionException(e);
```

Comment:
Not sure if it actually adds any value to wrap the caught runtime exception into a JobExecutionException (which is a Quartz exception - even though Quartz is not used here).

But yeah this class seems to be pretty much a direct copy from the old codebase so I don't really have an opinion whether or not this (whole class) should be refactored in any way

Reply:
You're right using Quartz exception does not make sense here since Quartz is not used anymore. This was copied from the current implementation. I changed this so that the exception is logged and then rethrown. Refactoring the class in the future is probably a good idea. In this case my goal was to replicate the current functionality with the minimum amount of modifications. It is probably easier to first replicate the current functionality more or less "as-is", and refactor it later once it's proven to be working.